File: N2.5/N2.5.js
function wordConverter(fourthBlank) {
    const convertedWord = [];
    let blankWord = fourthBlank;
    blankWord = blankWord.split('').sort().join('');
    convertedWord.push(blankWord);
    while (nextPermutation(blankWord)) {
      convertedWord.push(blankWord);
    }
    return convertedWord;
  }
  
  function nextPermutation(word) {
    let i = word.length - 2;
    while (i >= 0 && word[i] >= word[i + 1]) {
      i--;
    }
    if (i < 0) {
      return false;
    }
    let j = word.length - 1;
    while (word[j] <= word[i]) {
      j--;
    }
    swap(word, i, j);
    reverse(word, i + 1);
    return true;
  }
  
  function swap(word, i, j) {
    const temp = word[i];
    word[i] = word[j];
    word[j] = temp;
  }
  
  function reverse(word, start) {
    let i = start;
    let j = word.length - 1;
    while (i < j) {
      swap(word, i, j);
      i++;
      j--;
    }
  }
  
  function stringSplitter(enterString) {
    const splittedString = [];
    let miniWord = '';
    for (let i = 0; i < enterString.length; i++) {
      if (enterString[i] === ' ') {
        splittedString.push(miniWord);
        miniWord = '';
      } else if (i === enterString.length - 1) {
        miniWord += enterString[i];
        splittedString.push(miniWord);
      } else {
        miniWord += enterString[i];
      }
    }
    return splittedString;
  }
  
  function translate(wordToTranslate, dictionary) {
    let translatedWord = '';
    for (let j = 0; j < wordToTranslate.length; j++) {
      const fifthBlank = wordToTranslate.substr(j, 1);
      translatedWord += dictionary[fifthBlank];
    }
    return translatedWord;
  }
  
  function unicChecker(translatedAllCombinations, rnComb) {
    let scoreOfSames = 0;
    for (const word of translatedAllCombinations) {
      if (rnComb === word) {
        scoreOfSames += 1;
      }
    }
    if (scoreOfSames > 1) {
      return '';
    } else {
      return rnComb;
    }
  }
  
  function main() {
    const dictionary = {
      a: '.-',
      b: '-...',
      w: '.--',
      g: '--.',
      d: '-..',
      e: '.',
      v: '...-',
      z: '--..',
      i: '..',
      j: '.---',
      k: '-.-',
      l: '.-..',
      m: '--',
      n: '-.',
      o: '---',
      p: '.--.',
      r: '.-.',
      s: '...',
      t: '-',
      u: '..-',
      f: '..-.',
      h: '....',
      c: '-.-.',
      q: '--.-',
      y: '-.--',
      x: '-..-',
    };
  
    let checkOfGo = 1;
    while (checkOfGo) {
      const enterString = prompt('Enter String - ');
  
      const splittedString = stringSplitter(enterString);
  
      let boolCheckOfWordLength = 1;
      for (const thirdBlank of splittedString) {
        if (thirdBlank.length > 9) {
          boolCheckOfWordLength = 0;
          break;
        }
      }
  
      let boolCheckOfStringLength = 1;
      if (enterString.length > 100 || enterString.length < 1) {
        boolCheckOfStringLength = 0;
      }
  
      if (!boolCheckOfStringLength || !boolCheckOfWordLength) {
        console.log('Please, enter correct string');
      } else {
        const allCombinations = [];
        for (const fourthBlank of splittedString) {
          for (const combination of wordConverter(fourthBlank)) {
            allCombinations.push(combination);
          }
        }
  
        const translatedAllCombinations = [];
        for (const wordToTranslate of allCombinations) {
          translatedAllCombinations.push(translate(wordToTranslate, dictionary));
        }
  
        const unicCombinations = new Set();
        for (const rnComb of translatedAllCombinations) {
          const sixBlank = unicChecker(translatedAllCombinations, rnComb);
          if (sixBlank === '') {
            continue;
          } else {
            unicCombinations.add(sixBlank);
          }
        }
  
        console.log('\nAll combinations and their Morze translation:');
        for (let outputCount = 0; outputCount < allCombinations.length; outputCount++) {
          console.log(`${allCombinations[outputCount]} = (${translatedAllCombinations[outputCount]})`);
        }
  
        console.log('Unic words:');
        for (const output of unicCombinations) {
          console.log(`(${output})`);
        }
  
        console.log('____________________________________________________');
        console.log(`Value of unic words - ${unicCombinations.size}`);
      }
      checkOfGo = 0;
    }
  }
  
  main();

File: N2.5/N.2.5.cs
using System;
using System.Collections.Generic;
using System.Linq;

namespace WordTranslator
{
    class Program
    {
        static List<string> WordConverter(string fourthblank)
        {
            List<string> ConvertedWord = new List<string>();
            char[] BlankWord = fourthblank.ToCharArray();
            Array.Sort(BlankWord);
            ConvertedWord.Add(new string(BlankWord));
            while (NextPermutation(BlankWord))
            {
                ConvertedWord.Add(new string(BlankWord));
            }
            return ConvertedWord;
        }

        static bool NextPermutation(char[] arr)
        {
            int i = arr.Length - 2;
            while (i >= 0 && arr[i] >= arr[i + 1])
            {
                i--;
            }
            if (i < 0)
            {
                return false;
            }
            int j = arr.Length - 1;
            while (arr[j] <= arr[i])
            {
                j--;
            }
            Swap(arr, i, j);
            Array.Reverse(arr, i + 1, arr.Length - i - 1);
            return true;
        }

        static void Swap(char[] arr, int i, int j)
        {
            char temp = arr[i];
            arr[i] = arr[j];
            arr[j] = temp;
        }

        static List<string> StringSplitter(string EnterString)
        {
            List<string> SplittedString = new List<string>();
            string[] words = EnterString.Split(' ');
            foreach (string word in words)
            {
                SplittedString.Add(word);
            }
            return SplittedString;
        }

        static string Translate(string WordToTranslate, Dictionary<string, string> Dictionary)
        {
            string TranslatedWord = "";
            foreach (char c in WordToTranslate)
            {
                string fifthblank = c.ToString();
                TranslatedWord += Dictionary[fifthblank];
            }
            return TranslatedWord;
        }

        static string UnicChecker(List<string> TranslatedAllCombinations, string RNComb)
        {
            int SOS = 0; //ScoreOfSames
            foreach (string word in TranslatedAllCombinations)
            {
                if (RNComb == word)
                {
                    SOS += 1;
                }
            }
            if (SOS > 1)
            {
                return "";
            }
            else
            {
                return RNComb;
            }
        }

        static void Main(string[] args)
        {
            int checkofgo = 1;
            while (checkofgo == 1)
            {
                Dictionary<string, string> Dictionary = new Dictionary<string, string>()
                {
                    {"a", ".-"}, {"b", "-..."}, {"w", ".--"}, {"g", "--."}, {"d", "-.."},
                    {"e", "."}, {"v", "...-"}, {"z", "--.."}, {"i", ".."}, {"j", ".---"},
                    {"k", "-.-"}, {"l", ".-.."}, {"m", "--"}, {"n", "-."}, {"o", "---"},
                    {"p", ".--."}, {"r", ".-."}, {"s", "..."}, {"t", "-"}, {"u", "..-"},
                    {"f", "..-."}, {"h", "...."}, {"c", "-.-."}, {"q", "--.-"}, {"y", "-.--"}, {"x", "-..-"}
                };

                Console.Write("Enter String - ");
                string EnterString = Console.ReadLine();

                List<string> SplittedString = StringSplitter(EnterString);

                int BoolCheckOfWordLength = 1;
                foreach (string thirdblank in SplittedString)
                {
                    if (thirdblank.Length > 9)
                    {
                        BoolCheckOfWordLength = 0;
                        break;
                    }
                }

                int BoolCheckOfStringLength = 1;
                if (EnterString.Length > 100 || EnterString.Length < 1)
                {
                    BoolCheckOfStringLength = 0;
                }

                if (BoolCheckOfStringLength == 0 || BoolCheckOfWordLength == 0)
                {
                    Console.WriteLine("Please, enter correct string");
                }
                else
                {
                    List<string> AllCombinations = new List<string>();
                    foreach (string fourthblank in SplittedString)
                    {
                        foreach (string Combination in WordConverter(fourthblank))
                        {
                            AllCombinations.Add(Combination);
                        }
                    }
                    List<string> TranslatedAllCombinations = new List<string>();
                    foreach (string WordToTranslate in AllCombinations)
                    {
                        TranslatedAllCombinations.Add(Translate(WordToTranslate, Dictionary));
                    }
                    HashSet<string> UnicCombinations = new HashSet<string>();
                    foreach (string RNComb in TranslatedAllCombinations)
                    {
                        string sixblank = UnicChecker(TranslatedAllCombinations, RNComb);
                        if (sixblank == "")
                        {
                            continue;
                        }
                        else
                        {
                            UnicCombinations.Add(sixblank);
                        }
                    }
                    Console.WriteLine();
                    Console.WriteLine("All combinations and their Morze translation: ");
                    for (int outputcount = 0; outputcount < AllCombinations.Count; outputcount++)
                    {
                        Console.WriteLine(AllCombinations[outputcount] + " = " + "(" + TranslatedAllCombinations[outputcount] + ")");
                    }
                    Console.WriteLine("Unic words: ");
                    foreach (string Output in UnicCombinations)
                    {
                        Console.Write("(" + Output + ")" + " ");
                    }
                    Console.WriteLine();
                    Console.WriteLine("____________________________________________________");
                    Console.WriteLine("               " + "Value of unic words - " + UnicCombinations.Count);
                }
                checkofgo = 0;
            }
        }
    }
}